### ISSUE TITLE
Load Pre-training Backbone

### ISSUE THREAD

**Issue by Lightmannnn** [ultralytics/ultralytics#17520]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have a backbone weight for yolov8m trained using a self-supervised network, and I've found in the course of fine-tuning on downstream tasks using yolov8 that this pre-training weight yields virtually unchanged performance whether loaded or not. Here's how I implemented it
`model = YOLO('yolov8m-seg.yaml')`
`model.model.load_state_dict(torch.load("myweight.pth")['state_dict'], strict=False)`
`model.train(...)`

AND in ultralytics\engine\model.py, I commented out line 799, replacing it with
`self.trainer.model = self.model ` to prevent recreating the model

I don't know what the problem is. Can you help me?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Lightmannnn, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help direct you towards resources that may assist you further, but rest assured an Ultralytics engineer will follow up with you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com) site, which offers a wealth of information for new users, including various examples of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. You might find answers to many common questions there.

From your description, it seems like you’re facing an issue with loading pre-trained weights and observing expected results. If this is a 🐛 Bug Report, please ensure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in reproducing the issue effectively.

For questions related to custom training or implementation, we suggest you provide any relevant dataset image examples and training logs, and confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Also, feel free to join the Ultralytics community discussions where you can connect with other users:
- For interactive discussions, you can join our [Discord](https://ultralytics.com/discord) 🎧.
- For detailed discussions, visit our [Discourse](https://community.ultralytics.com) page.
- You can also check out and contribute to our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are using the latest `ultralytics` package by upgrading along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) using a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in several up-to-date, verified environments, pre-installed with necessary dependencies like [CUDA](https://developer.nvidia.com/cuda), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Refer to [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

When this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. The CI tests validate the operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and with each commit.

We appreciate your patience and support!

**Comment by Y-T-G** [comment]
> AND in ultralytics\engine\model.py, I commented out line 799, replacing it with
> `self.trainer.model = self.model ` to prevent recreating the model
> 

Did you make the changes in the package installation folder?

What's the training code?

**Comment by glenn-jocher** [comment]
@Lightmannnn it seems like you've made the necessary changes to load your custom backbone weights. Ensure that the weights are correctly aligned with the model architecture. If the performance remains unchanged, consider verifying the compatibility of your pre-trained weights with the YOLOv8 architecture and checking for any discrepancies in the dataset or training configuration.
